src/types.c

```diff
@@ -164,20 +164,12 @@ reset_ip_rule(NetplanIPRule* ip_rule)
 static void
 reset_backend_settings(NetplanBackendSettings* settings, NetplanBackend backend)
 {
-    switch (backend) {
-        case NETPLAN_BACKEND_NETWORKD:
-            FREE_AND_NULLIFY(settings->networkd.unit);
-            break;
-        case NETPLAN_BACKEND_NM:
-            FREE_AND_NULLIFY(settings->nm.name);
-            FREE_AND_NULLIFY(settings->nm.uuid);
-            FREE_AND_NULLIFY(settings->nm.stable_id);
-            FREE_AND_NULLIFY(settings->nm.device);
-            g_datalist_clear(&settings->nm.passthrough);
-            break;
-        default:
-            break;
-    }
+    FREE_AND_NULLIFY(settings->networkd.unit);
+    FREE_AND_NULLIFY(settings->nm.name);
```

Comment:
I was pondering if this could lead to double-frees or corrupt data, as `backend_settings` is a union of `struct NetplanNMSettings` and `struct NetplanNetworkdSettings`, therefore `settings->networkd.unit` and `settings.nm->name` are basically the same pointer (both `char *`).

But we're nullifying the pointer, so the `FREE_AND_NULLIFY(settings->nm.name);` call is basically a no-op here. But we should probably it as-is for clarity.

Looking further.. I don't think `backend_settings->networkd.unit` is being used anywhere. I wonder why it's there at all?? Maybe we should just replace that union with just the `NetplanNMSettings` struct, which should be ABI-safe, as we won't change memory size or layout. We could then add an additional `NetplanNetworkdSettings` struct in the future (separately, not as a union), should we ever need it. This should make the situation much clearer. WDYT`

Reply:
I did just that, it made things a bit simpler.